plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/transition/MapboxViewportTransitionFactory.kt

```diff
@@ -0,0 +1,278 @@
+        )
+
+      val centerAnimationRate = 500L
+      centerDuration =
+        ((screenDistanceFromMapCenterToLocation / centerAnimationRate) * 1000L).toLong()
+          .coerceAtMost(MAXIMUM_LOW_TO_HIGH_DURATION_MS)
+      animators += createCenterAnimator(
+        center = center,
+        duration = centerDuration
+      )
+    }
+
+    var zoomDelay = 0L
+    var zoomDuration = 0L
+    cameraOptions.zoom?.let { zoom ->
+      val currentMapCameraZoom = currentMapCameraState.zoom
+      val zoomDelta = abs(zoom - currentMapCameraZoom)
+      val zoomAnimationRate = 2.2
+      zoomDelay = centerDuration / 2L
+      zoomDuration = ((zoomDelta / zoomAnimationRate) * 1000L).toLong()
+        .coerceAtMost(MAXIMUM_LOW_TO_HIGH_DURATION_MS)
+
+      animators += createZoomAnimator(
+        zoom = zoom,
+        startDelay = zoomDelay,
+        duration = zoomDuration
+      )
+    }
+
+    cameraOptions.bearing?.let { bearing ->
+      val bearingDuration = 1800L
+      val bearingDelay = (zoomDelay + zoomDuration - bearingDuration).coerceAtLeast(0L)
+      animators += createBearingAnimator(
+        rotation = normalizeBearing(
+          currentMapCameraState.bearing,
+          bearing
+        ),
+        startDelay = bearingDelay,
+        duration = bearingDuration
+      )
+    }
+
+    val pitchAndPaddingDuration = 1200L
+    val pitchAndPaddingDelay = (zoomDelay + zoomDuration - pitchAndPaddingDuration + 100L)
+      .coerceAtLeast(0L)
+    cameraOptions.pitch?.let { pitch ->
+      animators += createPitchAnimator(
+        pitch = pitch,
+        startDelay = pitchAndPaddingDelay,
+        duration = pitchAndPaddingDuration
+      )
+    }
+
+    cameraOptions.padding?.let { padding ->
+      animators += createPaddingAnimator(
+        padding = padding,
+        startDelay = pitchAndPaddingDelay,
+        duration = pitchAndPaddingDuration
+      )
+    }
+
+    return createAnimatorSet(animators).constrainDurationTo(transitionMaxDurationMs)
+  }
+
+  fun transitionFromHighZoomToLowZoom(
+    cameraOptions: CameraOptions,
+    transitionMaxDurationMs: Long
+  ): AnimatorSet {
+    val animators = mutableListOf<Animator>()
+    val currentMapCameraState = cameraManager.cameraState
+
+    cameraOptions.center?.let { center ->
+      animators += createCenterAnimator(
+        center = center,
+        startDelay = 800L,
+        duration = 1000L
+      )
+    }
+
+    cameraOptions.zoom?.let { zoom ->
+      animators += createZoomAnimator(
+        zoom = zoom,
+        duration = 1800L
+      )
+    }
+
+    cameraOptions.bearing?.let { bearing ->
+      animators += createBearingAnimator(
+        rotation = normalizeBearing(
+          currentMapCameraState.bearing,
+          bearing
+        ),
+        startDelay = 600L,
+        duration = 1200L
+      )
+    }
+
+    cameraOptions.pitch?.let { pitch ->
+      animators += createPitchAnimator(
+        pitch = pitch,
+        duration = 1000L
```

Comment:
we can use `LINEAR_ANIMATION_DURATION_MS` here, or probably change the name to `ANIMATION_DURATION_MS` so it can be used for all the constants.

Reply:
i see @kiryldz already commented above the same, removing it.